gpkit/nomial_data.py

```diff
@@ -54,6 +45,24 @@ def __hash__(self):
             self._hashvalue = hash(tuple(zip(self.exps, self.cs)))
         return self._hashvalue
 
+    @classmethod
+    def fromnomials(cls, nomials):
+        """Construct a NomialData from an iterable of Signomial objects"""
+        nd = cls()  # use pass-through version of __init__
```

Comment:
modifying which object in `fromnomials`? I need to create one, which is the purpose of calling `cls`. We also need `init_from_monomials` so that children (GeometricProgram, in this case) can call `super.init_from_monomials`. (I don't see an obvious way for subclassers to call the `fromnomials` classmethod).

Does that answer what you were asking?